packages/graphql-lowcode/src/generate/hasura/hasuraUtils.ts

```diff
@@ -0,0 +1,54 @@
+import { Argument, TypesObject, Field } from '../types'
+import { getNestedOfType } from '../generateGraphqlQueries'
+
+export function getHasuraInputFields(args: Argument[]): String[] {
+  const containsWhere = args.some(argument => argument.name === 'where')
+  const containsId = args.some(argument => argument.name === 'id')
+  const containsObject = args.some(argument => argument.name === 'object')
+  const containsObjects = args.some(argument => argument.name === 'objects')
+  const containsSet = args.some(argument => argument.name === '_set')
+  const containsPKColumns = args.some(argument => argument.name === 'pk_columns')
+
+  if (containsObjects) return ['objects'] //insert many
+  if (containsObject) return ['object'] //insert one
+
+  if (containsSet && containsWhere) return ['where', '_set'] //update many
+  if (containsSet && containsPKColumns) return ['_set', 'pk_columns'] //update one
+
+  if (containsWhere) return ['where'] //delete many
+  return ['id'] //delete one
+}
+
+export function buildReturningString(types: TypesObject[], returningType: string): string {
+  let returningFields: string[] = []
+
+  for (const type of types) {
```

Comment:
use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find instead for loop for better readability
```
const type = types.find(type => type.name === returningType && type.fields)
const scalarFields =  type.fields.filter(field => field..kind === SCALAR)
const returningFields = scalarFields.map(field => field.name)
```